packages/insea-connect-backend/src/main/java/ma/insea/connect/chat/group/GroupService.java

```diff
@@ -49,8 +54,12 @@ public List<Group> findallgroupsofemail(Long myId) {
         return groups;
     }
     public void deleteGroup(Long groupId) {
-        Group group = groupRepository.findById(groupId).get();
-        groupRepository.delete(group);
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        User user = userRepository.findByUsername(authentication.getName()).orElse(null);
+        if (user.getId() == groupRepository.findById(groupId).get().getCreator()){
+            Group group = groupRepository.findById(groupId).get();
```

Comment:
extract groupRepository.findById(...) to a variable, so you don't fire up the same request to the database

Reply:
done